src/csm_testing/tests/iscsi_prechecks/__main__.py

```diff
@@ -0,0 +1,321 @@
+#!/usr/bin/env python3
+#
+# MIT License
+#
+# (C) Copyright 2025 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+"""
+This script validates run iSCSI PRE-CHECKS.
+"""
+
+import os
+import sys
+import re
+import json
+from csm_testing.lib.iscsi_common import (
+    run_command,
+    compare_versions,
+    iscsi_prechecks_failed,
+    get_compute_node,
+    get_worker_node,
+    check_iscsid,
+    check_multipathd,
+)
+
+
+def check_craycli_minimum_version(minimum_version: str) -> None:
+    """
+    Function to check craycli version and print the info.
+    Args:
+        minimum_version(str)
+    Returns:
+        None
+    """
+    craycli_current_version, returncode = run_command(
+        command="rpm -qa | grep craycli | sed 's/^[a-zA-Z.-]*-\\([0-9][^ ]*\\)-.*$/\\1/'"
```

Comment:
This can be simplified
```suggestion
        command="rpm -q --qf '%{VERSION}' craycli"
```

Reply:
Impleted in check_craycli_minimum_version, check_cli_minimum_version_xname in iscsci_common.py